src/services/jsTyping.ts

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the Apache License, Version 2.0. 
+// See LICENSE.txt in the project root for complete license information.
+
+/// <reference path='services.ts' />
+
+/* @internal */
+namespace ts.JsTyping {
+
+    interface HostType {
+        directoryExists: (path: string) => boolean;
+        fileExists: (fileName: string) => boolean;
+        readFile: (path: string, encoding?: string) => string;
+        readDirectory: (path: string, extension?: string, exclude?: string[], depth?: number) => string[];
+        writeFile: (path: string, data: string, writeByteOrderMark?: boolean) => void;
+    };
+
+    var _host: HostType;
+    var _safeList: Map<string>;
+
+    // a typing name to typing file path mapping
+    var inferredTypings: Map<string> = {};

```

Comment:
This is dangerously structured also. You are storing this value on the namespace (i.e. a singleton per server/language service) and accessing it from several top-level functions, yet the entries it contains are specific to a project.